metadata-ingestion/src/datahub/ingestion/source/ldap.py

```diff
@@ -387,8 +392,10 @@ def build_corp_user_mce(
 
         manager_urn = f"urn:li:corpuser:{manager_ldap}" if manager_ldap else None
 
+        user_urn = email if self.config.use_email_as_username else ldap_user
```

Comment:
what happens if email is None?

Reply:
email is rendered via `get_attr_or_none` . default is `full_name` for group &  `ldap_user' for user, if value for `email`  key is absent in `attrs` dict.

```
email = get_attr_or_none(attrs, self.config.user_attrs_map["email"], ldap_user)
```